config/peerpods/podvm/gcp-podvm-image-handler.sh

```diff
@@ -0,0 +1,261 @@
+# Create image (-c)
+# Delete image (-C)
+
+[[ "$DEBUG" == "true" ]] && set -x
+
+# include common functions from lib.sh
+# shellcheck source=/dev/null
+# The directory is where gcp-podvm-image-handler.sh is located
+source "$(dirname "$0")"/lib.sh
+
+# Function to verify that the required variables are set
+
+function verify_vars() {
+  # Ensure CLOUD_PROVIDER is set to gcp
+  [[ -z "${CLOUD_PROVIDER}" || "${CLOUD_PROVIDER}" != "gcp" ]] && error_exit "CLOUD_PROVIDER is empty or not set to gcp"
+
+  required_vars=(
+    # From gcp-podvm-image-cm:
+    "INSTALL_PACKAGES"
+    "DISABLE_CLOUD_CONFIG"
+
+    # From lib.sh:
+    "CAA_SRC_DIR"
+  )
+
+  for var in "${required_vars[@]}"; do
+    [[ -z "${!var}" ]] && error_exit "$var is not set or empty"
+  done
+}
+
+# function to download and install gcloud cli
+
+function install_gcloud_cli() {
+  # Install gcloud cli
+  # If any error occurs, exit the script with an error message
+
+  # Check if gcloud cli is already installed
+  if command -v gcloud &>/dev/null; then
+    echo "gcloud cli is already installed"
+    return
+  fi
+
+  # Download gcloud cli
+  # TODO
+  tee -a /etc/yum.repos.d/google-cloud-sdk.repo <<EOM
+[google-cloud-cli]
+name=Google Cloud CLI
+baseurl=https://packages.cloud.google.com/yum/repos/cloud-sdk-el9-x86_64
+enabled=1
+gpgcheck=1
+repo_gpgcheck=0
+gpgkey=https://packages.cloud.google.com/yum/doc/rpm-package-key.gpg
+EOM
+
+  dnf install -y google-cloud-cli
+
+  echo "GCP CLI installed successfully"
+}
+
+# Function to recreate podvm-images configmap with all the images
+# TODO: Not sure if aws, azure is working either.
+function recreate_image_configmap() {
+  echo "Recreating podvm-images configmap not implemented"
+}
+
+# Creates a GCP Image with the PodVM distro.
+function create_image() {
+  echo "Creating GCP Image"
+  # Create the GCP image
+  # If any error occurs, exit the script with an error message
+
+  # Install packages if INSTALL_PACKAGES is set to yes
+  if [[ "${INSTALL_PACKAGES}" == "yes" ]]; then
+    # Install required rpm packages
+    install_rpm_packages
+
+    # Install required binary packages
+    install_binary_packages
+  fi
+
+  # Based on the value of `IMAGE_TYPE` the image is either build from scratch or
+  # usingthethe prebuilt artifact.
+  if [[ "${IMAGE_TYPE}" == "operator-built" ]]; then
+    error_exit "Currently only pre-built is supported for GCP, exiting."
+  elif [[ "${IMAGE_TYPE}" == "pre-built" ]]; then
+    create_image_from_prebuilt_artifact
+  fi
+
+  # Get the image id of the newly created image.
+  # This will set the IMAGE_ID variable
+  get_image_id
+
+  # Add the image id as annotation to peer-pods-cm configmap
+  update_cm_annotation "LATEST_IMAGE_ID" "${IMAGE_ID}"
+
+  echo "GCP image created successfully"
+}
+
+function create_image_from_prebuilt_artifact() {
+  echo "Creating GCP image from prebuilt artifact"
+
+  # Set the IMAGE_VERSION and IMAGE_NAME
+  set_image_version_and_name
```

Comment:
This `set_image_version_and_name` function is not implemented.

Reply:
During rebase, I dropped by mistake, thanks.